packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import { CAROUSEL_ITEM } from './constants';
+import { useCarouselCollection_unstable } from './useCarouselCollection';
+import { useCarouselWalker_unstable } from './useCarouselWalker';
+import { isHTMLElement, useMergedRefs } from '@fluentui/react-utilities';
+
+// TODO: Migrate this into an external @fluentui/carousel component
+// For now, we won't export this publicly, is only for internal TeachingPopover use until stabilized.
+export function useCarousel_unstable() {
+  const { ref: carouselRef, walker: carouselWalker } = useCarouselWalker_unstable();
+  const { store, value, setValue, currentIndex, setIndex, totalPages } = useCarouselCollection_unstable();
+
+  const rootRef = React.useRef<HTMLDivElement>(null);
+
+  React.useEffect(() => {
+    const allItems = rootRef.current?.querySelectorAll(`[${CAROUSEL_ITEM}]`)!;
+
+    for (let i = 0; i < allItems.length; i++) {
+      store.addValue(allItems.item(i).getAttribute(CAROUSEL_ITEM)!);
+    }
+  }, [store]);
```

Comment:
```suggestion
  React.useEffect(() => {
    const allItems = rootRef.current?.querySelectorAll(`[${CAROUSEL_ITEM}]`)!;

    for (let i = 0; i < allItems.length; i++) {
      store.addValue(allItems.item(i).getAttribute(CAROUSEL_ITEM)!);
    }
    
    return () => {
       store.clear();
    }
  }, [store]);
```

We need to call `store.clear()` there, otherwise in Strict Mode the set of items will be doubled.